storybookjs/storybook
# UI: Improve tabs component, more type correct, allow for `FC` as title

This targets `release/7.2` to not disrupt the `7.1` release.

## What I did

- improve the Tabs component to have better types
- the title can be a `FC` now
- improve the Addon interface to accept the `FC` as title as well
- split the Addon interface into `Addon_BaseType` for future use

## How to test

- build should succeed
- yarn check should pass
- no UI changes detected by chromatic (except for new stories)
- there should be no perf changes in anything related to tabs or addons

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
